src/gf.c

```diff
@@ -1420,6 +1420,12 @@ JL_DLLEXPORT void jl_method_table_insert(jl_methtable_t *mt, jl_method_t *method
             (jl_tupletype_t*)type, simpletype, jl_emptysvec, (jl_value_t*)method, 0, &method_defs,
             method->min_world, ~(size_t)0, &oldvalue);
     if (oldvalue) {
+        if (oldvalue == (jl_value_t*)method) {
+            // redundant add of same method; no need to do anything
```

Comment:
Sounds like a bug elsewhere? There should be a one-to-one relation between jl_new_method and jl_method_table_insert to avoid corruption.

Reply:
I thought that might be the case. There were definitely two references to the same method in the list argument to `jl_insert_methods`. We can look into that. I still think this change makes sense as a defensive measure though.